include/kodlab_mjbots_sdk/joint_moteus.h

```diff
@@ -103,129 +114,193 @@ class JointMoteus: public JointBase
-            LOG_WARN("With the current moteus kp of %.3f, the max value of the joint kp is %.3f", moteus_kp_, kp_/kp_scale);
-          }
-          return kp_scale;
-        }
+        [[nodiscard]] float get_kp_scale() const;
 
         /*!
          * @brief accessor for kd_scale
          * @return kd_scale
          */
-        [[nodiscard]] float get_kd_scale() const {
-          if(moteus_kd_ == 0){
-            LOG_WARN("Moteus kd is set to 0, while attempting to send pd commands");
-            return 0;
-          }
-          const float kd_scale = kd_/moteus_kd_ * 2 * M_PI/gear_ratio_;
-          if(kd_scale > 1){
-            LOG_WARN("kd_scale is greater than 1, will be limited to 1. Either use a lower kd or increase kd in the moteus");
-            LOG_WARN("With the current moteus kd of %.3f, the max value of the joint kd is %.3f", moteus_kp_, kd_/kd_scale);
-          }
-          return kd_scale;
-        }
+        [[nodiscard]] float get_kd_scale() const;
 
         /*!
          * @brief accessor for the moteus position target
          * @return the moteus position target in units of revolutions
          */
-        [[nodiscard]] float get_moteus_position_target()const{return (position_target_ + zero_offset_) * gear_ratio_/direction_/2/M_PI;}
+        [[nodiscard]] float get_moteus_position_target()const;
 
         /*!
          * @brief accessor for the moteus velocity target
          * @return the moteus velocity target in units of revolutions/s
          */
-        [[nodiscard]] float get_moteus_velocity_target()const{return (velocity_target_) * gear_ratio_/direction_/2/M_PI;}
-
+        [[nodiscard]] float get_moteus_velocity_target()const;
 
     private:
         int can_id_;   /// the can id of this joint's moteus
         int can_bus_;  /// the can bus the moteus communicates on
+        ::mjbots::moteus::QueryCommand query_type_;
         ::mjbots::moteus::Mode mode_ = ::mjbots::moteus::Mode::kStopped; /// joint's moteus mode
         float moteus_kp_ = 0;
         float moteus_kd_ = 0;
+        float temperature_ = 0;
+        float q_current_ = 0;
+        float d_current_ = 0;
+        float voltage_ = 0;
+        ::mjbots::moteus::Fault fault_ = ::mjbots::moteus::Fault::kSuccess; // fault mode
+};
+
+/**
+ * @brief struct for storing or passing moteus joint configurations
+ * 
+ */
+struct MoteusJointConfig{
+    int can_id;
+    int can_bus;
+    ::mjbots::moteus::QueryCommand query_type = JointMoteus::kDefaultQuery;
+    std::string name = "";
+    int direction = 1; 
+    float zero_offset = 0;
+    float gear_ratio = 1.0;
+    float max_torque = std::numeric_limits<float>::infinity();
+    float pos_min = -std::numeric_limits<float>::infinity();
+    float pos_max = std::numeric_limits<float>::infinity();
+    float moteus_kp = 0;
+    float moteus_kd = 0;
+    float soft_start_duration_ms = 1;
 };
+
```

Comment:
Why white space change?

Reply:
Followed formatting used in `moteus_protocol.h` and `joint_base.h`, but notice now that other files (such as `robot_base.h`) do not have whitespace here. Google Style guide on [namespace formatting](https://google.github.io/styleguide/cppguide.html#Namespace_Formatting) seems to suggest a line of vertical whitespace in this context, although the style guide also recommends limiting vertical whitespace when possible. 

Fine with either decision.

Reply:
Either is fine. Generally isolated white space changes are frowned upon.